kotlinx-coroutines-core/common/src/internal/LimitedDispatcher.kt

```diff
@@ -34,10 +35,10 @@ internal class LimitedDispatcher(
     private val workerAllocationLock = SynchronizedObject()
 
     @ExperimentalCoroutinesApi
-    override fun limitedParallelism(parallelism: Int): CoroutineDispatcher {
+    override fun limitedParallelism(parallelism: Int, name: String?): CoroutineDispatcher {
         parallelism.checkParallelism()
-        if (parallelism >= this.parallelism) return this
-        return super.limitedParallelism(parallelism)
+        if (parallelism >= this.parallelism) return namedOrThis(name)
+        return super.limitedParallelism(parallelism, name)
```

Comment:
(Minor) Since `checkParallelism` was already called, we could avoid calling `super` here, instead directly constructing a `LimitedDispatcher`.

Reply:
In things like that, I prefer leaving the double-check, but ensuring that the code is reused and has less entry point